multicluster/controllers/multicluster/serviceexport_controller.go

```diff
@@ -245,15 +260,24 @@ func (r *ServiceExportReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		}
 	}
 
-	ep := &corev1.Endpoints{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      req.Name,
-			Namespace: req.Namespace,
-		},
+	if r.endpointType == "ClusterIP" {
+		ep.Subsets = []corev1.EndpointSubset{common.GetServiceEndpointSubset(svc)}
+	} else {
+		if epInstalled {
+			installedEp := epObj.(*epInfo)
+			if apiequality.Semantic.DeepEqual(getEndPointsPorts(ep), installedEp.ports) &&
```

Comment:
Could not we call FilterEndpointSubsets() and compare ep.Subnets? And for `installedEp/epInfo`, could not we save  ep.Subnets directly?

I would avoid calling funcs like getEndPointsAddress() multiple times, which iterate and copy slices.

Reply:
Yes, refined it to save subsets directly. thanks.